src/PageHeader/PageHeader.tsx

```diff
@@ -0,0 +1,376 @@
+import React from 'react'
+import {Box} from '..'
+import {useResponsiveValue, ResponsiveValue} from '../hooks/useResponsiveValue'
+import {SxProp, merge, BetterSystemStyleObject} from '../sx'
+import Heading from '../Heading'
+import {ArrowLeftIcon} from '@primer/octicons-react'
+import Link from '../Link'
+import {ForwardRefComponent as PolymorphicForwardRefComponent} from '../utils/polymorphic'
+
+const REGION_ORDER = {
+  ContextArea: 0,
+  TitleArea: 1,
+  Description: 2,
+  Navigation: 3
+}
+
+// Types that are shared between sub components
+export type sharedPropTypes = {
+  visible?: boolean | ResponsiveValue<boolean>
+} & SxProp
+
+// Default state for the `visible` prop when a sub component is only visible on narrow viewport
+const onlyVisibleOnNarrowView = {
+  narrow: true,
+  regular: false,
+  wide: false
+}
+
+// Default state for the `visible` prop when a sub component is visible on regular and wide viewport
+const visibleOnRegularView = {
+  narrow: false,
+  regular: true,
+  wide: true
+}
+
+// Root
+// -----------------------------------------------------------------------------
+export type PageHeaderProps = {
+  'aria-label'?: React.AriaAttributes['aria-label']
+  as?: React.ElementType
+} & sharedPropTypes
+
+const Root: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({children, sx = {}}) => {
+  const rootStyles = {
+    display: 'flex',
+    flexDirection: 'column',
+    gap: '8px',
+    // 24px for wide and regular, 16px for narrow viewports
+    padding: useResponsiveValue(
+      {
+        narrow: 3,
+        regular: 4,
+        wide: 4
+      },
+      false
+    )
+  }
+  return <Box sx={merge<BetterSystemStyleObject>(rootStyles, sx)}>{children}</Box>
```

Comment:
Would we ever need to support an `as` prop in order to support semantic landmarks like `<header>` or is the intent that this would not be a landmark on the page?

Reply:
I think that makes sense. PageHeader can be a part of PageHeader (it could even be most of the time) but I agree, we should give an option where it needs to be the `header` landmark. Thanks!